webpack/assets/javascripts/react_app/components/common/table/index.js

```diff
@@ -0,0 +1,40 @@
+import { Table as PfTable } from 'patternfly-react';
+import React from 'react';
+import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
+import EmptyState from '../emptyState';
+
+export const headerFormat = value => <PfTable.Heading>{value}</PfTable.Heading>;
+export const cellFormat = value => <PfTable.Cell>{value}</PfTable.Cell>;
+export const ellipsisFormat = value => (
+  <PfTable.Cell>
+    <EllipsisWithTooltip>{value}</EllipsisWithTooltip>
+  </PfTable.Cell>
+);
+
+class Table extends React.Component {
+  isEmpty() {
+    return this.props.rows.length === 0;
+  }
+
+  render() {
+    const { columns, rows, emptyState } = this.props;
+    return this.isEmpty() ? (
+      <EmptyState {...emptyState} />
+    ) : (
+      <PfTable.PfProvider
```

Comment:
Hi @amirfefer, this looks good but what if I want to use `components` to `<PfTable.PfProvider>` - I am not sure I can do that through `Table`. In `Hardware Models` I am using it to set how the heading and cells will look like.

Reply:
This has been solved in Katello: https://github.com/Katello/katello/blob/master/webpack/move_to_foreman/components/common/table/components/Table.js That component is based on @amirfefer 's solution and is going to be moved to foreman soon. AFAIR the interfaces should be compatible.
So, if you don't mind @ripcurld, I wouldn't block merging this PR by that. It contains lot of other good stuff, has been open for quite some time and is quite ready to be merged.

Reply:
@tstrachota I didn't mean to block :smiley:  but just to raise a concern I have in [`Hardware Models`](https://github.com/theforeman/foreman/pull/5755).